sample/src/main/java/com/canhub/cropper/sample/camera/presenter/SCameraPresenter.kt

```diff
@@ -92,41 +86,30 @@ internal class SCameraPresenter : SCameraContract.Presenter {
         view?.showErrorMessage("onCancel")
     }
 
-    override fun onActivityResult(resultCode: Int, requestCode: Int, data: Intent?) {
-        if (resultCode == RESULT_OK) {
-            when (requestCode) {
-                CropImage.CROP_IMAGE_ACTIVITY_REQUEST_CODE -> {
-                    val bitmap = context?.let { CropImage.getActivityResult(data)?.getBitmap(it) }
-                    Log.v(
-                        "File Path",
-                        context
-                            ?.let { CropImage.getActivityResult(data)?.getUriFilePath(it) }
-                            .toString()
-                    )
-
-                    CropImage.getActivityResult(data)?.uriContent?.let {
-                        view?.handleCropImageResult(it.toString().replace("file:", ""))
-                    } ?: view?.showErrorMessage("CropImage getActivityResult return null")
-                }
-                SCameraFragment.CUSTOM_REQUEST_CODE -> {
-                    context?.let {
-                        Log.v("File Path", CropImage.getPickImageResultUriFilePath(it, data))
-                        CropImage.getPickImageResultUriFilePath(it, data)
-                        val uri = CropImage.getPickImageResultUriContent(it, data)
-                        view?.handleCropImageResult(uri.toString().replace("file:", ""))
-                    }
-                }
-                CropImage.PICK_IMAGE_CHOOSER_REQUEST_CODE -> {
-                    context?.let { ctx ->
-                        Log.v("File Path", CropImage.getPickImageResultUriFilePath(ctx, data))
-                        val uri = CropImage.getPickImageResultUriContent(ctx, data)
-
-                        view?.handleCropImageResult(uri.toString())
-                    }
-                }
-                CODE_PHOTO_CAMERA -> view?.startCropImage(CameraEnumDomain.START_WITH_URI)
-                else -> view?.showErrorMessage("requestCode = $requestCode")
-            }
-        } else view?.showErrorMessage("resultCode = $resultCode")
+    override fun onCropImageResult(result: CropImageView.CropResult) {
+        if (result.isSuccessful) {
+            view?.handleCropImageResult(result.uriContent.toString().replace("file:", ""))
+        } else if (result is CropImage.CancelledResult) {
+            view?.showErrorMessage("cropping image was cancelled by the user")
+        } else {
+            view?.showErrorMessage("cropping image failed")
+        }
+    }
+
+    override fun onPickImageResult(resultUri: Uri?) {
+        if (resultUri != null) {
+            Log.v("File Path", resultUri.toString())
```

Comment:
I just realise we are not returning `UriFilePath` just `UriContent`. Will add a comment into `PickImageContract` class